packages/mui-base/src/utils/appendOwnerState.ts

```diff
@@ -12,12 +12,14 @@ export default function appendOwnerState(
   existingProps: Record<string, any> = {},
   ownerState: object,
 ) {
+  const { ownerState: existingPropsOwnerState, ...otherExistingProps } = existingProps;
```

Comment:
@michaldudak what do you think of this change? For example, the core components could send `ownerState` to extend the one generated from the base components, would be nice if we don't need to add the check for whether is host component over there as well. It would mean however that we mutate (or we will need to set `existingProps.ownerState` to `undefined`)